kotlinx-coroutines-core/common/test/flow/operators/FlowOnTest.kt

```diff
@@ -261,6 +261,21 @@ class FlowOnTest : TestBase() {
         finish(3)
     }
 
+    @Test
+    fun testCancellation() = runTest {
+        val result = flow {
+            emit(1)
+            emit(2)
+            emit(3)
+            expectUnreached()
+            emit(4)
+        }.flowOn(wrapperDispatcher())
+            .buffer(0)
```

Comment:
This `buffer(0)` shall be removed. Otherwise, this test does not fail without the fix. Alternatively, a test that check for cancellation with default buffer size shall be used here.